components/director/internal/domain/formationtemplate/repository.go

```diff
@@ -81,12 +93,37 @@ func (r *repository) Get(ctx context.Context, id string) (*model.FormationTempla
 	return result, nil
 }
 
-// GetByName returns a single FormationTemplate by given name
-func (r *repository) GetByName(ctx context.Context, templateName string) (*model.FormationTemplate, error) {
-	log.C(ctx).Debugf("Getting formation template by name: %q...", templateName)
+// GetByNameAndTenant returns a single FormationTemplate by given name and tenant ID
+func (r *repository) GetByNameAndTenant(ctx context.Context, templateName, tenantID string) (*model.FormationTemplate, error) {
+	log.C(ctx).Debugf("Getting formation template by name: %q and tenant %q ...", templateName, tenantID)
 	var entity Entity
-	if err := r.singleGetterGlobal.GetGlobal(ctx, repo.Conditions{repo.NewEqualCondition("name", templateName)}, repo.NoOrderBy, &entity); err != nil {
-		return nil, err
+
+	conditionsEqualName := repo.Conditions{repo.NewEqualCondition("name", templateName)}
+	conditionsEqualNameAndNullTenant := repo.Conditions{
+		repo.NewEqualCondition("name", templateName),
+		repo.NewNullCondition(tenantIDColumn),
+	}
+
+	// If the call is with tenant but the query (select * from FT where name = ? and tenant_id = ?) returns NOT FOUND that means that there is no such tenant scoped FT and the call should get the global FT with that name.
+	//
+	// With this approach we allow the client to create a tenant scoped FT with the same name as a global FT - so when we get the FT first we will try to match it by name and tenant
+	// and if there is no such FT, we will get the global one
+	if tenantID == "" {
+		if err := r.singleGetterGlobal.GetGlobal(ctx, conditionsEqualNameAndNullTenant, repo.NoOrderBy, &entity); err != nil {
+			return nil, err
+		}
+	} else {
+		err := r.singleGetterWithEmbeddedTenant.Get(ctx, resource.FormationTemplate, tenantID, conditionsEqualName, repo.NoOrderBy, &entity)
```

Comment:
You can make this `err..` one-liner. But if you think it will be less readable you can leave it this way.